test/upgrade/upgrade.go

```diff
@@ -54,3 +58,18 @@ func assertServiceResourcesUpdated(t pkgTest.TLegacy, clients *test.Clients, nam
 		t.Fatal(fmt.Sprintf("The endpoint for Route %s at %s didn't serve the expected text %q: %v", names.Route, url, expectedText, err))
 	}
 }
+
+func createNewService(serviceName string, t *testing.T) {
```

Comment:
This looks very similar to what `TestHelloWorld` does. Is it necessary to test for creating a new service after upgrade and downgrade?

Reply:
Good question. I think the `TestHelloworld` test doesn't currently run after an upgrade/downgrade. It only runs on a fresh new installation. I'd say this test is good to have because upgrading Knative is not 100% same as installing a fresh new version. Please correct me if I'm wrong. 
This test might also be beneficial for downstream where upgrades might be done in a slightly different way than how it is done upstream. In some cases upgrades will certainly be slightly different from installing a fresh new version of Knative.
I'd like to hear other opinions on this topic though.

Reply:
I see your point. I'm not familiar with the different cases upgrades will be slightly different. Would love to hear people's opinions.